### PR TITLE
Add consensus pulled per txn counter along with bucket information

### PR DESCRIPTION
**Pull request by vusirikala**
## Description
Add consensus pulled per txn counter along with bucket information

## Type of Change
- [ ] New feature
- [ ] Bug fix
- [ ] Breaking change
- [ ] Performance improvement
- [ ] Refactoring
- [ ] Dependency update
- [ ] Documentation update
- [ ] Tests

## Which Components or Systems Does This Change Impact?
- [ ] Validator Node
- [ ] Full Node (API, Indexer, etc.)
- [ ] Move/Aptos Virtual Machine
- [ ] Aptos Framework
- [ ] Aptos CLI/SDK
- [ ] Developer Infrastructure
- [ ] Other (specify)

## Checklist
- [ ] I have read and followed the [CONTRIBUTING](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md) doc
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I identified and added all stakeholders and component owners affected by this change as reviewers
- [ ] I tested both happy and unhappy path of the functionality
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: mempool/src/core_mempool/mempool.rs [hunk 1]
```diff
@@ -155,6 +155,7 @@ impl Mempool {
                 priority.as_str(),
             );
             counters::CORE_MEMPOOL_TXN_CONSENSUS_PULLED.observe((prev_count + 1) as f64);
+            counters::CORE_MEMPOOL_TXN_CONSENSUS_PULLED_BY_BUCKET.with_label_values(bucket.as_str()).observe((prev_count + 1) as f64);
```
Comment (bchocho) [inline mempool/src/core_mempool/mempool.rs:158]: Let's just replace `CORE_MEMPOOL_TXN_CONSENSUS_PULLED`?

Comment (vusirikala) [reply mempool/src/core_mempool/mempool.rs:158]: If we do that, then the forge tests will have the "bucket" parameter, whereas the current devnet, testnet and mainnet won't have this parameter. So, the grafana plots could get confusing for a while?

Comment (vusirikala) [reply mempool/src/core_mempool/mempool.rs:158]: If that's not an issue, then I can just change the `CORE_MEMPOOL_TXN_CONSENSUS_PULLED` counter.

Comment (bchocho) [reply mempool/src/core_mempool/mempool.rs]: Yeah I think we can get around this issue by just having one dashboard for each, then remove the old one once this is fully rolled out

File: mempool/src/counters.rs [hunk 2]
```diff
@@ -310,7 +310,7 @@ pub static CORE_MEMPOOL_TXN_CONSENSUS_PULLED: Lazy<Histogram> = Lazy::new(|| {
     register_histogram!(
         "aptos_core_mempool_txn_consensus_pulled",
         "Number of times a txn was pulled from core mempool by consensus",
-        TXN_CONSENSUS_PULLED_BUCKETS.to_vec()
+        &["bucket"]
```
Comment (bchocho) [inline mempool/src/counters.rs:313]: You'll need to replace `register_histogram` with `register_histogram_vec`

And let's keep the histogram buckets override as the numbers should be better than the default (not to be confused with the bucket label)

```
register_histogram_vec!(..., &["bucket"], TXN_CONSENSUS_PULLED_BUCKETS.to_vec())

Comment (vusirikala) [reply mempool/src/counters.rs]: Done.

File: mempool/src/counters.rs [hunk 3]
```diff
@@ -307,9 +307,10 @@ pub static CORE_MEMPOOL_GC_LATENCY: Lazy<HistogramVec> = Lazy::new(|| {
 });
 
 pub static CORE_MEMPOOL_TXN_CONSENSUS_PULLED: Lazy<Histogram> = Lazy::new(|| {
-    register_histogram!(
+    register_histogram_vec!(
```
Comment (ibalajiarun) [inline mempool/src/counters.rs:310]: Suggest renaming the metric too. No idea how our TSDB would react to this migration.

Comment (vusirikala) [reply mempool/src/counters.rs:310]: Changed the name.

**Review by igor-aptos — APPROVED** [review]
you can update the dashboard to show both metrics in the same chart. then it will work both before and after this change. and we can clean it up later

**Review by bchocho — APPROVED** [review]

**Pull request merged**